src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -99,36 +125,81 @@ private void ReplayErrors(AggregateException exceptions)
 
         private void CheckPackageVulnerabilities(IReadOnlyList<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>> knownVulnerabilities)
         {
-            Dictionary<PackageIdentity, Dictionary<PackageVulnerabilityInfo, List<string>>>? packagesWithKnownVulnerabilities =
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            Dictionary<PackageIdentity, PackageAuditInfo>? packagesWithKnownVulnerabilities =
                 FindPackagesWithKnownVulnerabilities(knownVulnerabilities);
+            stopwatch.Stop();
+            CheckPackagesDurationSeconds = stopwatch.Elapsed.TotalSeconds;
 
             if (packagesWithKnownVulnerabilities == null) return;
 
+            stopwatch.Restart();
+
+            int directPackageCount = packagesWithKnownVulnerabilities.Values.Count(p => p.IsDirect);
+            DirectPackagesWithAdvisory = new(capacity: directPackageCount);
+            TransitivePackagesWithAdvisory = new(capacity: packagesWithKnownVulnerabilities.Count - directPackageCount);
```

Comment:
do you account for the mode here?

Reply:
In your review comment you wrote:

> Do we get transitive data if the mode is direct?
> Can we maybe do that?

By not accounting for the mode setting, we always collect transitive package data 😆

Reply:
Tricky, tricky :D